src/main/java/uk/gov/ons/census/casesvc/cache/UacQidCache.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.ons.census.casesvc.cache;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.casesvc.client.UacQidServiceClient;
+import uk.gov.ons.census.casesvc.model.dto.UacQidDTO;
+
+@Component
+public class UacQidCache {
+  private final UacQidServiceClient uacQidServiceClient;
+
+  @Value("${uacservice.uacqid-cache-min}")
+  private int cacheMin;
+
+  @Value("${uacservice.uacqid-fetch-count}")
+  private int cacheFetch;
+
+  @Value("${uacservice.uacqid-get-timeout}")
+  private long uacQidGetTimout;
+
+  private static Executor executor = Executors.newFixedThreadPool(8);
```

Comment:
you can make this final. I know how much you love to make things final

Reply:
finally